### ISSUE TITLE
can't updade, can't uninstall

### ISSUE THREAD

**Issue by hr1232** [FriendsOfShopware/FroshPlatformMailArchive#58]
### PHP Version

8.2.7-1~deb12u1

### Shopware Version

6.5.1.0

### Actual behaviour

Plugin version 2.0.2  is installed

Attempt to install update ends in error message:
`Can not download plugin. Please contact your system administrator. Error: can not downloads plugins managed by composer from store api`

Attemp to uninstall ends in error message:
`An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails`

### Expected behaviour

I just want to uninstall this fucked up situation and forget ever having had this addon.

### Steps to Reproduce?

- install version 2.0.2
- try to uninstall/update

**Comment by hr1232** [comment]
Additional information:

I ran this to migrate storage into sql:
```
./bin/console frosh:mailarchive:migrate
 3/3 [============================] 100%
```

This didn't change anything, error messages still persist. Cannot update and cannot uninstall. How can I get rid of this fucked up plugin???

**Comment by hr1232** [comment]
```
php bin/console plugin:uninstall mailarchive

Shopware Plugin Lifecycle Service
=================================

 Uninstall 1 plugin(s):
 * Mail Archive (v2.0.2)

21:04:19 CRITICAL  [console] Error thrown while running command "plugin:uninstall mailarchive". Message: "An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails" ["exception" => Doctrine\DBAL\Exception\ForeignKeyConstraintViolationException^ { …},"command" => "plugin:uninstall mailarchive","message" => "An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails"]

In ExceptionConverter.php line 56:

  An exception occurred while executing a query: SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails


In Exception.php line 28:

  SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails


In Connection.php line 32:

  SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails


plugin:uninstall [-r|--refresh] [-c|--clearCache] [--skip-asset-build] [--keep-user-data] [--] <plugins>...
```

**Comment by schneider-felix** [comment]
Plugin uninstall will be fixed with #54 

As for your other problem, it looks like you are trying to update the plugin using the administration but it's actually installed using composer.
